tests/test_functions.py

```diff
@@ -745,8 +754,47 @@ def case(arg1, arg2, expect_ret):  # type: (str, str, str | None) -> str
     with pytest.raises(RuntimeError, match="offset too large"):
         case("swapfrom", "10", "<unfinished>")
 
+
 def test51_isinstance():
     assert t.isinstance_(3, int)
     assert not t.isinstance_(3, bool)
     with pytest.raises(TypeError):
         t.isinstance_(3, 7)
+
+
+def test52_bool():
+    # straight up bool
+    assert t.test_11_b(True) is True
+    assert t.test_11_b(False) is False
+    assert t.test_11_bnc(True) is True
+    assert t.test_11_bnc(False) is False
+
+    # None requires implicit conversion
+    with pytest.raises(TypeError):
+        assert t.test_11_bnc(None)
+    # t.test_11_b(None) is False
```

Comment:
I couldn't get the `None` case to work directly, it seems we don't end up in the type caster for `bool`, but didn't debug further. Maybe someone with better understanding of the code paths, can quickly spot the problem, otherwise I could spend more time debugging.

Reply:
You'd need `nb::arg().none()` in test_11_b, but perhaps wait for Wenzel's opinion on enabling such conversion.